Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -1409,6 +1409,10 @@ WebIDL::ExceptionOr<void> Navigable::navigate(NavigateParams params)
                 }
                 finalize_a_cross_document_navigation(*this, to_history_handling_behavior(history_handling), history_entry);
             });
+
+            if (is_top_level_traversable() && url != history_entry->url) {
```

Comment:
when we append session history traversal steps to the traversal navigable, we actually put the navigation change into a queue to be executed later and merged with any other pending navigations (with the option for the user/javascript to cancel the navigation, for e.g.). 

At the very least, we should try to move this check into the traversal steps above, just after finalizing the cross document navigation. Though, I'm not sure if we should be doing this inside that finalize_a_cross_document_navigation() function anyway? to keep the URL sync'd at all times.

@kalenikaliaksandr what do you think?